augur/export_v2.py

```diff
@@ -874,7 +902,15 @@ def parse_node_data_and_metadata(T, node_data_files, metadata_file):
                 node_attrs[name][corrected_key] = value
                 node_data_names.add(corrected_key)
 
-    return (node_data, node_attrs, node_data_names, metadata_names)
+    # third pass: create `branch_attrs` which includes certain traits supplied in `node_attrs`
+    # (e.g. mutations are coverted to branch attrs, and `clade_annotation` is interpreted as a label)
+    # as well as any branch labels supplied in node-data files.
+    branch_attrs = {}
+    transfer_mutations_to_branches(node_attrs, branch_attrs)
+    transfer_clade_annotation_to_branches(node_attrs, branch_attrs)
+    transfer_branch_labels_to_branch_attrs(node_data.get('branch_labels', {}), branch_attrs)
```

Comment:
This is exactly the pattern and implementation [I was recommending in the schema updates PR](https://github.com/nextstrain/augur/pull/727#discussion_r644430884)! I love it! 😄

Reply:
Ha! I still prefer closures in some cases tho 😂